presto-main/src/main/java/com/facebook/presto/operator/window/LagFunction.java

```diff
@@ -52,14 +57,22 @@ public void processRow(BlockBuilder output, int frameStart, int frameEnd, int cu
 
             long valuePosition = currentPosition - offset;
 
-            if ((valuePosition >= 0) && (valuePosition <= currentPosition)) {
-                windowIndex.appendTo(valueChannel, Ints.checkedCast(valuePosition), output);
-            }
-            else if (defaultChannel >= 0) {
-                windowIndex.appendTo(defaultChannel, currentPosition, output);
-            }
-            else {
-                output.appendNull();
+            while (true) {
```

Comment:
This control flow is hard to follow. Break out the null skipping logic to happen before the append logic.